foxpuppet/windows/browser/bookmarks/bookmark.py

```diff
@@ -0,0 +1,212 @@
+            self.selenium.find_element(*BookmarkLocators.OTHER_BOOKMARKS_STAR).click()
+            self.selenium.find_element(*BookmarkLocators.SAVE_BUTTON).click()
+
+    def add_bookmark(self, bookmark_data: BookmarkData) -> None:
+        """Add a Bookmark using the main bookmark menu."""
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.actions.context_click(
+                self.selenium.find_element(*BookmarkLocators.NAVIGATOR_TOOLBOX)
+            ).perform()
+            self.selenium.find_element(*BookmarkLocators.MENU_BAR).click()
+            self.selenium.find_element(*BookmarkLocators.MAIN_MENU_BOOKMARK).click()
+            self.actions.context_click(
+                self.selenium.find_element(*BookmarkLocators.MANAGE_BOOKMARKS)
+            ).perform()
+            self.selenium.find_element(*BookmarkLocators.ADD_BOOKMARK).click()
+
+            frame_element = self.selenium.find_element(
+                *BookmarkLocators.ADD_BOOKMARK_FRAME
+            )
+            if frame_element is not None:
+                self.selenium.switch_to.frame(frame_element)
+            else:
+                raise NoSuchElementException(
+                    f"iFrame with locator {BookmarkLocators.ADD_BOOKMARK_FRAME} not found."
+                )
+
+            if bookmark_data["name"]:
+                self.actions.send_keys(bookmark_data["name"]).perform()
+            self.actions.send_keys(Keys.TAB).perform()
+
+            if bookmark_data["url"]:
+                self.actions.send_keys(bookmark_data["url"] + Keys.TAB).perform()
+
+            tags = bookmark_data["tags"]
+            if tags is not None:
+                for tag in tags:
+                    self.actions.send_keys(tag).perform()
+                    self.actions.send_keys(",").perform()
+                self.actions.send_keys(Keys.TAB).perform()
+
+            if bookmark_data.get("keyword"):
+                keyword = bookmark_data["keyword"] or ""
+                self.actions.send_keys(keyword + Keys.TAB).perform()
+
+            self.actions.send_keys(Keys.TAB, Keys.TAB, Keys.TAB, Keys.ENTER).perform()
+            self.actions.send_keys(Keys.TAB, Keys.ENTER).perform()
+
+            self.selenium.switch_to.default_content()
+
+    def retrieve_bookmark(self, label: str) -> bool:
+        """
+        Check if a bookmark with the given label exists.
+
+        Args:
+            label (str): The name of the bookmark to search for.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.selenium.find_element(*BookmarkLocators.PANEL_MENU).click()
+            self.selenium.find_element(*BookmarkLocators.PANEL_BOOKMARK_MENU).click()
+            panel_bookmarks = self.selenium.find_element(
+                *BookmarkLocators.PANEL_BOOKMARK_TOOLBAR
+            )
+            if panel_bookmarks:
+                menu_items = panel_bookmarks.find_elements(
```

Comment:
Why do you check if the `panel_bookmarks` is true? Wouldn't selenium throw an error if it couldn't find the element needed?